packages/core/src/domain/configuration/endpointBuilder.ts

```diff
@@ -31,9 +31,8 @@ export function createEndpointBuilder(
 ) {
   const { clientToken } = initConfiguration
 
-  const host = buildEndpointHost(initConfiguration, endpointType)
-  const baseUrl = `https://${host}/api/v2/${INTAKE_TRACKS[endpointType]}`
-  const proxyUrl = initConfiguration.proxyUrl && normalizeUrl(initConfiguration.proxyUrl)
+  const baseUrl = buildEndpointBaseUrl(initConfiguration, endpointType)
+  const proxyUrl = buildProxyUrl(initConfiguration)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏IMO the overall implementation could be easier to follow by being more explicit about the different cases that we are handling.

Reply:
I reworked things a bit. Let me know if it's clearer.

Reply:
LGTM!